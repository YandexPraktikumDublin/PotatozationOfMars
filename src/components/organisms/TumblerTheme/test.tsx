import React from 'react'
import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { TumblerTheme } from '.'

describe('<TumblerTheme />', () => {
  const darkThemeClass = 'dark'

  it('should renders correct <TumblerTheme />', () => {
    const wrapper = shallow(<TumblerTheme />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should toggle theme', () => {
    const wrapper = mount(<TumblerTheme />)

    expect(wrapper.find('img').prop('style')).toEqual({
      top: 'calc(50% - 12.5px)',
      transform: 'translateX(-2px)'
    })

    expect(
      document.documentElement.classList.contains(darkThemeClass)
    ).toBeTruthy()

    wrapper.simulate('click')

    expect(wrapper.find('img').prop('style')).toEqual({
      top: 'calc(50% - 12.5px)',
      transform: 'translateX(100%)'
    })

    expect(
      document.documentElement.classList.contains(darkThemeClass)
    ).toBeFalsy()

    wrapper.simulate('click')

    expect(wrapper.find('img').prop('style')).toEqual({
      top: 'calc(50% - 12.5px)',
      transform: 'translateX(-2px)'
    })

    expect(
      document.documentElement.classList.contains(darkThemeClass)
    ).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
