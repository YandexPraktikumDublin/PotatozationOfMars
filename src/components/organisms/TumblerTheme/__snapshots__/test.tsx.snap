// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TumblerTheme /> should renders correct <TumblerTheme /> 1`] = `
<button
  className="relative bg-primary border-2 border-white rounded-2xl w-12	h-5 outline-none"
  onClick={[Function]}
>
  <img
    alt="Toggle theme"
    className="absolute transition ease-in-out w-6 h-6"
    height="24"
    src={[Function]}
    style={
      Object {
        "top": "calc(50% - 0.75rem)",
        "transform": "translateX(-2px)",
      }
    }
    width="24"
  />
</button>
`;
